File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    for (const [i, target] of goal.entries()) {
        if (cards1[0] === target) {
            cards1.shift(); 
        } else if (cards2[0] === target) {
            cards2.shift();
        } else {
            return "No"; 
        }  
    }
    return "Yes";
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {           
        int curIdx1 = 0;
        int curIdx2 = 0;
        for (String target : goal) {
            if (curIdx1 < cards1.length && target.equals(cards1[curIdx1])) {
                curIdx1++;
            } else if (curIdx2 < cards2.length && target.equals(cards2[curIdx2])) {
                curIdx2++;
            } else {
                return "No";
            }
        }
        return "Yes";
    }
    
}